src/HomeScreen/index.jsx

```diff
@@ -36,12 +46,20 @@ const HomeScreen = React.memo(({ navigation }) => {
   };
   useEffect(() => {
     dispatch(actions.setLanguage(language));
+    const isValidBaniOrder =
+      baniOrder != null && // checks for null or undefined
+      typeof baniOrder === "object" &&
+      Array.isArray(baniOrder.baniOrder) &&
+      baniOrder.baniOrder.length > 0;
```

Comment:
We can convert this code to a one-liner
![image](https://github.com/user-attachments/assets/9565f44a-b9f7-4004-aeee-d843a259c313)